src/callbacks_step/amr.jl

```diff
@@ -852,6 +852,38 @@ function original2refined(original_cell_ids, refined_original_cells, mesh)
     return shifted_cell_ids[original_cell_ids]
 end
 
+# Auxiliary function to compute the new element ids for refined elements
+# Used when applying positivity limiter after refinement step
+# Saves every first id of the 2^ndims child elements
+# All child elements can be addressed with element_ids_new[i]:(element_ids_new[i] + 2^ndims - 1)
+function compute_new_ids_refined_elements(elements_to_refine, mesh)
+    element_ids_new = copy(elements_to_refine)
+    for i in eachindex(element_ids_new)
+        # Each refined element increases all ids of the following elements by 2^ndims - 1
+        for j in (i + 1):length(element_ids_new)
+            element_ids_new[j] += 2^ndims(mesh) - 1
+        end
+    end
+
+    return element_ids_new
+end
+
+# Auxiliary function to compute the new element ids for removed elements
+# Used when applying positivity limiter after coarsening step
+function compute_new_ids_removed_elements(elements_to_remove, mesh)
+    @assert length(elements_to_remove) % (2^ndims(mesh))==0 "The length of `elements_to_remove` must be a multiple of 2^ndims(mesh)."
+
+    element_ids_new = zeros(Int, div(length(elements_to_remove), 2^ndims(mesh)))
+    for i in eachindex(element_ids_new)
+        # New element id is the id of the first child
+        # Additionally, all following ids decrease by (2^ndims - 1) per coarsened element
+        element_ids_new[i] = elements_to_remove[2^ndims(mesh) * (i - 1) + 1] -
+                             (2^ndims(mesh) - 1) * (i - 1)
+    end
+
+    return element_ids_new
+end
+
```

Comment:
Before, this computation of the new element ids was done inside the limiter call. Therefore, it was done for every variable (density and pressure).
Now, I outsourced this computation to auxiliary functions. It is dimension independent.